### PR TITLE
Feature: APP-547 Fix token list on token menu

### PR DESCRIPTION
**Pull request by sepehr2github**
## Description

- Token fetching issue fixed on the token menu
-Custom contract checked on deposit flow and It's working fine

Task: [APP-547](https://aragonassociation.atlassian.net/browse/APP-547)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran all tests with success and extended them if necessary.
- [ ] I have updated the ``CHANGELOG.md`` file in the root folder.
- [x] I have tested my code on the test network.

### CODE REVIEW

File: packages/web-app/src/context/providers.tsx [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import {InfuraProvider, Web3Provider} from '@ethersproject/providers';
 import React, {createContext, useContext, useEffect, useState} from 'react';
-import {useWallet} from 'use-wallet';
+import {useWallet} from 'hooks/useWallet';
```
Comment (Fabricevladimir) [inline packages/web-app/src/context/providers.tsx:3]: Aha!

**Review by Fabricevladimir — APPROVED** [review]
Thanks. One comment though is that we probably shouldn't return tokens with balance 0 since the user cannot deposit them in the first place.

File: packages/web-app/src/context/providers.tsx [hunk 2]
```diff
@@ -31,10 +31,8 @@ type ProviderProviderProps = {
  * therefore be null if no wallet is connected.
  */
 export function ProvidersProvider({children}: ProviderProviderProps) {
-  const {chainId, ethereum} = useWallet();
-  const [web3Provider, setWeb3Provider] = useState(
-    ethereum ? new Web3Provider(ethereum) : null
-  );
+  const {chainId, provider} = useWallet();
+  const [web3Provider, setWeb3Provider] = useState(provider || null);
```
Comment (RakeshUP) [inline packages/web-app/src/context/providers.tsx]: do we need to store this in a separate state variable? can't we just use the provider returned by useWallet and send it to the child components?

**Review by RakeshUP — APPROVED** [review]

**Comment by sepehr2github** [comment]
> Thanks. One comment though is that we probably shouldn't return tokens with balance 0 since the user cannot deposit them in the first place.

That's true, but regardless when the user sees the insufficient balance, he/she will understand, anyway, good idea we can add later

**Pull request merged**